### PR TITLE
New contract updates (wrapper changes + misc)

### PR DESCRIPTION
**Pull request by TateB**
- inclusive of #49 
- adds support for value in textchanged event
- adds some fixes for namewrapping

### CODE REVIEW

File: schema.graphql [hunk 1]
```diff
@@ -1,18 +1,21 @@
 type Domain @entity {
-  id: ID!                                               # The namehash of the name
```
Comment (Arachnid) [inline schema.graphql:2]: Can you leave this formatting intact?

File: schema.graphql [hunk 2]
```diff
@@ -104,21 +137,38 @@ type NameTransferred implements RegistrationEvent @entity {
...
-  id: ID!                   # Concatenation of resolver address and namehash
+  id: ID! # Concatenation of resolver address and namehash
   domain: Domain
-  address: Bytes!           # Address of resolver contract
-
-  addr: Account             # Current value of addr record (per events)
-  contentHash: Bytes        # Content hash, in binary format.
-  texts: [String!]          # Set of observed text record keys
-  coinTypes: [BigInt!]      # Set of observed SLIP-44 coin types
+  address: Bytes! # Address of resolver contract
```
Comment (Arachnid) [inline schema.graphql:160]: Can you preserve this formatting please?

File: src/ethRegistrar.ts [hunk 3]
```diff
@@ -8,25 +8,26 @@ import {
...
 import {
   NameRegistered as NameRegisteredEvent,
   NameRenewed as NameRenewedEvent,
-  Transfer as TransferEvent,
+  Transfer as TransferEvent
 } from './types/BaseRegistrar/BaseRegistrar'
 
+import {
+  NameRegistered as ControllerNameRegisteredEventOld,
+} from './types/EthRegistrarControllerOld/EthRegistrarControllerOld'
```
Comment (Arachnid) [inline src/ethRegistrar.ts:22]: You should still listen for renew events as well; the old controller will not be deactivated immediately.

File: src/reverseRegistrar.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { ReverseClaimed } from "./types/ReverseRegistrar/ReverseRegistrar";
+import { Account, Domain, ReverseClaimed as ReverseClaimedEvent } from "./types/schema";
+import { createEventID } from './utils'
+
+export function handleReverseClaimed(event: ReverseClaimed): void {
+    const account = new Account(event.params.addr.toHex())
+    const domain = new Domain(event.params.node.toHex())
+    domain.save()
+    account.reverseDomain = domain.id
```
Comment (Arachnid) [inline src/reverseRegistrar.ts:9]: If we're doing this, we should a) Check if the name actually resolves (eg, if the forward resolution matches), and b) Invalidate it if the forward resolution changes.

**Review by Arachnid — CHANGES_REQUESTED** [review]

File: src/ethRegistrar.ts (hunk 3 above)
Comment (TateB) [reply src/ethRegistrar.ts:22]: Still listening for renew on the old controller, it's just the same event so they can both reference one.

**Review by Arachnid — APPROVED** [review]

**Pull request merged**